src/payment/strategies/cybersource/cybersource-payment-strategy.ts

```diff
@@ -0,0 +1,145 @@
+
+        if (!payment) {
+            throw new MissingDataError(MissingDataErrorType.MissingPayment);
+        }
+
+        if (!this._paymentMethod || !this._paymentMethod.config) {
+            throw new MissingDataError(MissingDataErrorType.MissingPaymentMethod);
+        }
+
+        return this._paymentMethod.config.is3dsEnabled ? this._placeOrderUsing3DS(order, payment, options, this._paymentMethod.clientToken) :
+            this._placeOrder(order, payment, options);
+    }
+
+    finalize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return Promise.reject(new OrderFinalizationNotRequiredError());
+    }
+
+    deinitialize(options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        return Promise.resolve(this._store.getState());
+    }
+
+    private _placeOrderUsing3DS(order: OrderRequestBody, payment: OrderPaymentRequestBody, options?: PaymentRequestOptions, clientToken?: string): Promise<InternalCheckoutSelectors> {
+        if (!clientToken) {
+            return Promise.reject(new MissingDataError(MissingDataErrorType.MissingPaymentMethod));
+        }
+
+        if (!payment.paymentData) {
+            return Promise.reject(new MissingDataError(MissingDataErrorType.MissingPayment));
+        }
+
+        const paymentData = payment.paymentData as CreditCardInstrument;
+
+        return this._cardinalClient.configure(clientToken)
+            .then(() => this._cardinalClient.runBindProcess(paymentData.ccNumber)
+            .then(() => {
+                return this._placeOrder(order, payment, options)
+                    .catch(error => {
+                        if (!(error instanceof RequestError) || !some(error.body.errors, { code: 'enrolled_card' })) {
+                            return Promise.reject(error);
+                        }
+
+                        return this._cardinalClient.getThreeDSecureData(error.body.three_ds_result, this._getOrderData(paymentData))
+                            .then(threeDSecure =>
+                                this._store.dispatch(this._paymentActionCreator.submitPayment({
+                                    ...payment,
+                                    paymentData: {
+                                        ...paymentData,
+                                        threeDSecure,
+                                    },
+                                }))
+                            );
+                    });
+            }));
+    }
+
+    private _placeOrder(order: OrderRequestBody, payment: OrderPaymentRequestBody, options?: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        if (!payment.paymentData) {
+            return Promise.reject(new MissingDataError(MissingDataErrorType.MissingPayment));
+        }
+
+        return this._store.dispatch(this._orderActionCreator.submitOrder(order, options))
+            .then(() =>
+                this._store.dispatch(this._paymentActionCreator.submitPayment({
```

Comment:
Do we really need to add `paymentData` again? it seems redundant.

```ts
{ ...payment, paymentData: payment.paymentData } === payment
```

am I missing something?